wordpress-mobile/WordPress-Android
# Fix crash in UploadStarter on Samsung devices with Android 5

Fixes #10827 

Fixes a crash on Samsung devices running Android 5. The app is crashing because it can't find `_state` variable in the runtime. However, we don't directly use this variable in our code. As it turned out the variable is used in `Mutex.kt`. This PR simply removes the usage of `Mutex` from our codebase and replaces it with `@Synchronized`.

More info about why the app is crashing can be find in @jd-alexander [comment](https://github.com/wordpress-mobile/WordPress-Android/issues/10827#issuecomment-557365282).

To test:
I don't have a Samsung phone running Android 5. But we should at least verify it still works as expected on other phones.
1. Turn off the internet connection
2. Create a post
3. Publish it
4. Turn on internet connection + pull to refresh
5. Make sure the post gets uploaded 

PR submission checklist:



Thank you @jd-alexander for the initial investigation which really helped me fix this issue!

@jkmassel We might want to consider merging this change as hotfix - wdyt? It crashed 3.6k times to 1.2k users in the last 90 days. I'm a bit worried that it might not work on Samsung devices running Android 5 at all - I haven't checked the stats though. I believe the change is very safe.
